hare/hare.go

```diff
@@ -156,20 +165,14 @@ var ErrTooLate = errors.New("consensus process finished too late")
 // records the provided output
 func (h *Hare) collectOutput(ctx context.Context, output TerminationOutput) error {
 	set := output.Set()
-	blocks := make([]types.BlockID, len(set.values))
-	i := 0
+	blocks := make([]types.BlockID, 0, len(set.values))
 	for v := range set.values {
-		blocks[i] = v
-		i++
-	}
-
-	// check validity of the collected output
-	if !h.validate(blocks) {
-		h.WithContext(ctx).Error("failed to validate the collected output set")
+		blocks = append(blocks, v)
```

Comment:
at least in older versions of go, creating the slice and assigning each index when the size is known was more efficient than append. though I get append is more readable.